mosviz/controls/tests/test_slit_controller.py

```diff
@@ -0,0 +1,228 @@
+import pytest
+
+import numpy as np
+
+from astropy.wcs import WCS
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+from astropy.wcs.utils import proj_plane_pixel_area
+
+from mosviz.viewers.mos_viewer import MOSVizViewer
+
+
+def construct_test_wcs(ra=0., dec=0., x=0., y=0., area=1.0):
+    """
+    Constructs an `~astropy.wcs.WCS` object according to params.
+    WCS object will always have ['RA---TAN', 'DEC--TAN'] transform.
+    :param ra: Center RA in deg
+    :param dec: Center DEC in deg
+    :param x: Center x pixel
+    :param y: Center Y pixel
+    :param area: ﻿Projected plane pixel area (deg**2/pix)
+    :return: `~astropy.wcs.WCS`
+    """
+    axis_scale = np.sqrt(area)
+    w = WCS()
+    w.wcs.crval = [ra, dec]
+    w.wcs.crpix = [x, y]
+    w.wcs.ctype = ['RA---TAN', 'DEC--TAN']
+    w.wcs.cdelt = np.array([-axis_scale, axis_scale])
+    return w
+
+
+def check_slits_and_patchs(slit_controller):
+    """Check if patch is correctly available"""
+    assert slit_controller.is_active == (slit_controller._patch is not None)
+
+    if slit_controller._pix_slit is not None and slit_controller._patch is None:
+        raise Exception("Slit Controller has _pix_slit but no patch.")
+    if slit_controller._slit is not None and slit_controller._patch is None:
+        raise Exception("Slit Controller has _slit but no patch.")
+
+    patch = slit_controller.patch
+    assert(patch is not None) == slit_controller.is_active
+
+
+def check_patch_attr(slit_controller, x, y, width, length):
+    """
+    Chcek the patch position, dimension and bounds.
+    Params are the expected values.
+    :param slit_controller: SlitController
+    :param x: correct center x pixel
+    :param y: correct center y pixel
+    :param width: correct width
+    :param length: correct length
+    """
+
+    x_bounds = np.array([x - (width / 2.), x + (width / 2.)])
+    y_bounds = np.array([y - (length / 2.), y + (length / 2.)])
+
+    assert np.isclose(slit_controller.x, x)
```

Comment:
assert are good for testing and should not be used to do standard checking. You should convert this to an if statement and then raise an Exception explicitly if any of these "fail".  There are some python optimization parameters that will (or used to) remove asserts from code so in those cases these checks will never happen.  I believe it would be best to leave asserts for the test infrastructure and use if/raise type of thing here.